src/Controller/Action/Admin/RenderPdfAction.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Controller\Action\Admin;
+
+use Gaufrette\Filesystem;
+use Knp\Bundle\SnappyBundle\Snappy\Response\PdfResponse;
+use Setono\SyliusGiftCardPlugin\Generator\GiftCardPdfPathGeneratorInterface;
+use Setono\SyliusGiftCardPlugin\Model\GiftCardConfigurationInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\HttpFoundation\HeaderUtils;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class RenderPdfAction
+{
+    private RepositoryInterface $giftCardConfigurationRepository;
+
+    private GiftCardPdfPathGeneratorInterface $giftCardPdfPathGenerator;
+
+    private Filesystem $filesystem;
+
+    public function __construct(
+        RepositoryInterface $giftCardConfigurationRepository,
+        GiftCardPdfPathGeneratorInterface $giftCardPdfPathGenerator,
+        Filesystem $filesystem
+    ) {
+        $this->giftCardConfigurationRepository = $giftCardConfigurationRepository;
+        $this->giftCardPdfPathGenerator = $giftCardPdfPathGenerator;
+        $this->filesystem = $filesystem;
+    }
+
+    public function __invoke(Request $request, int $id): PdfResponse
```

Comment:
I would have expected that the `$id` was the id of the gift card, but it's not apparently. Can you explain the flow of this. I don't seem to get it

Reply:
Well, we are in the admin, configuring a GiftCardConfiguration. We don't have any gift card right now (hence the dummy factory) but we just want to render the PDF that will be given to any user buying a gift card with this configuration

Reply:
Of course, haha. Well, the name of this action could be better, i.e. `RenderExamplePdfAction`